linera-ethereum/src/client.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+
+use ethers::{
+    prelude::{Http, Provider},
+    types::{Log, U256},
+};
+use ethers_core::types::{Address, BlockId, BlockNumber, Filter, U64};
+use ethers_middleware::Middleware;
+
+use crate::common::EthereumServiceError;
+
+pub struct EthereumEndpoint {
+    pub url: String,
+    pub provider: Provider<Http>,
+}
+
+impl EthereumEndpoint {
+    pub fn new(url: String) -> Result<Self, EthereumServiceError> {
+        let provider = Provider::<Http>::try_from(&url)?;
+        let provider = provider.interval(Duration::from_millis(10u64));
+        Ok(Self { url, provider })
+    }
+
+    pub async fn get_accounts(&self) -> Result<Vec<String>, EthereumServiceError> {
```

Comment:
If I'm reading the documentation correctly, this gets the accounts on the local node? So it wouldn't be something an oracle could use, would it?
We should at least document the public methods; it wasn't clear to me _which_ accounts would be returned.

Reply:
Indeed, that function is not included in my work on the Oracle done so far.